src/lucky_record/query_builder.cr

```diff
@@ -71,6 +72,12 @@ class LuckyRecord::QueryBuilder
     self
   end
 
+  def distinct_on(column)
+    @distinct = true
```

Comment:
Rather than setting @distinct to true, you could create a new private method called distinct? That returns true if @distinct is set or if @distinct_on is set. That way the logic for determining whether to add a distinct query is in one place